notebooks/02-ingestion/02-batch-ingestion.ipynb

```diff
@@ -0,0 +1,1008 @@
+   "metadata": {},
+   "source": [
+    "Data for both sources `kttm_data` and `country_lookup` are obtained from external sources:\n",
+    "```\n",
+    "WITH\n",
+    "kttm_data AS\n",
+    "(\n",
+    "  SELECT *\n",
+    "  FROM TABLE(\n",
+    "    EXTERN(\n",
+    "               '{\"type\":\"http\",\"uris\":[\"https://static.imply.io/example-data/kttm-v2/kttm-v2-2019-08-25.json.gz\"]}',\n",
+    "               '{\"type\":\"json\"}'\n",
+    "    )\n",
+    "  ) EXTEND (\"timestamp\" VARCHAR, \"agent_category\" VARCHAR, ...)\n",
+    "),\n",
+    "country_lookup AS\n",
+    "(\n",
+    "  SELECT *\n",
+    "  FROM TABLE(\n",
+    "    EXTERN(\n",
+    "      '{\"type\":\"http\",\"uris\":[\"https://static.imply.io/example-data/lookup/countries.tsv\"]}',\n",
+    "      '{\"type\":\"tsv\",\"findColumnsFromHeader\":true}'\n",
+    "    )\n",
+    "  ) EXTEND (\"Country\" VARCHAR, \"Capital\" VARCHAR, \"ISO3\" VARCHAR, \"ISO2\" VARCHAR)\n",
+    ")\n",
+    "```\n",
+    "\n",
+    "Columns from both tables can be used in the SELECT expressions using the alias `country_lookup` to reference any joined column:\n",
+    "```\n",
+    "  kttm_data.\"country\",\n",
+    "  country_lookup.\"Capital\" AS \"capital\",\n",
+    "  country_lookup.\"ISO3\" AS \"iso3\"\n",
+    "```\n",
+    "\n",
+    "The join is specified in the FROM clause:\n",
+    "```\n",
+    "FROM kttm_data\n",
+    "LEFT JOIN country_lookup ON country_lookup.Country = kttm_data.country\n",
+    "```\n",
+    "LEFT JOIN insured that we get all the rows from `kttm_data` source, an INNER JOIN would exclude rows from `kttm_data` if the value for `kttm_data.country` is not present in `country_lookup.Country`. \n",
+    "Since we did not set any context parameters, the join is processed as a broadcast join. The first table in the FROM clause is the distributed table and all other joined tables will be shipped to the workers to execute the join.\n",
+    "\n",
+    "Take a look at the data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "01fb13e7-9071-46b6-8088-810c72257073",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "druid.display.sql(\"\"\"\n",
+    "SELECT\n",
+    "  \"iso3\" AS \"country_code\", \n",
+    "  \"capital\",\n",
+    "  count( DISTINCT \"ip_address\" ) distinct_users, \n",
+    "  MIN(\"session_length\")/1000 fastest_session_ms,\n",
+    "  MAX(\"session_length\")/1000 slowest_session_ms\n",
+    "FROM \"kttm_enhanced\"\n",
+    "WHERE \"event_type\"='LayerClear'\n",
+    "GROUP BY 1,2\n",
+    "ORDER BY 3 DESC\n",
+    "LIMIT 10\n",
+    "\"\"\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "caa697e8-bec5-4d77-a424-5ee37a86b859",
+   "metadata": {},
+   "source": [
+    "##### Shuffle Joins - Large Lookup to Fact Joins\n",
+    "\n",
+    "Druid 26.0.0 introduced [shuffle joins with SQL Based Ingestion](https://druid.apache.org/docs/latest/multi-stage-query/reference.html#sort-merge).\n",
```

Comment:
As people are already using 26, I think you can drop this.

Reply:
done